jvm/src/main/kotlin/app/cash/trifle/internal/validators/CertChainValidatorFactory.kt

```diff
@@ -2,16 +2,29 @@ package app.cash.trifle.internal.validators
 
 import app.cash.trifle.Certificate
 import app.cash.trifle.Certificate.Companion.CERTIFICATE_VERSION
+import java.util.Date
 
 /**
  * Factory class that determines a specific certificate chain validator from the certificate
  * anchor's version.
  */
 object CertChainValidatorFactory {
-  fun get(certAnchor: Certificate): CertChainValidator {
+
+  /**
+   * Return a certificate validator matching the provided certAnchor.
+   * @param certAnchor - The certificate to use as the root certificate in the certificate chain. Its
+   *   format will determine how verification should be performed.
+   * @param date - The date to use for verification against certificates' validity windows. If null,
+   *   the current time is used.
+   */
+  fun get(certAnchor: Certificate, date: Date?): CertChainValidator {
```

Comment:
just curious when a date passed into as a parameter would be more useful than the system time?

Reply:
the obvious need here is for testing, but I also didn't see any reason not to let clients specify it for whatever purpose.